src/Controller/Controller.php

```diff
@@ -159,8 +163,9 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      * after action logic.
      *
      * @var bool
+     * @deprecated 3.6.0 Use enableAutoRender()/disableAutoRender() and isAutoRenderEnabled() instead.
      */
-    public $autoRender = true;
+    protected $autoRender = true;
```

Comment:
Will this be a BC issue from components setting this on the controller?

Reply:
With this it should be ok:
https://github.com/robertpustulka/cakephp/blob/e2d894e65b7731b971a8ded78d42971bb32b7c27/src/Controller/Controller.php#L403-L407